code/src/sixsq/nuvla/server/resources/api_docs/openapi.clj

```diff
@@ -0,0 +1,122 @@
+                         (when (scrud-op-supported? :get)
+                           (resource-get resource-metadata)))
+        collection-ops (merge
+                         (when (scrud-op-supported? :query)
+                           (collection-query resource-metadata)))]
+    (merge
+      (when (seq collection-ops)
+        {(str "/" type-uri) collection-ops})
+      (when (seq resource-ops)
+        {(str "/" type-uri "/{" resource-param "}")
+         (merge
+           {:parameters [{:name        resource-param
+                          :in          "path"
+                          :required    true
+                          :description (str "The ID of the " resource-type-name)
+                          :schema      {:type   "string"
+                                        :format "uuid"}}]}
+           resource-ops)}))))
+
+
+(defn paths
+  "Returns the `paths` section of the OpenAPI spec."
+  [resources-metadata]
+  (reduce (fn [m resource-metadata]
+            (merge m (resource-paths resource-metadata)))
+          (sorted-map)
+          resources-metadata))
+
+
+(defn schemas
+  "Returns the `schemas` section of the OpenAPI spec."
+  [resources-metadata]
+  (->> resources-metadata
+       (map openapi-schema/->openapi-schemas)
+       (apply merge)
+       (into (sorted-map))))
+
+
+(defn openapi-spec
+  "Returns the OpenAPI spec for the given resource types."
+  [resources-metadata]
+  (let [nuvla-api-spec {:openapi      "3.0.3"
+                        :info         {:title          "Nuvla API"
+                                       :description    "HTTP-based RESTful API, for the management of Nuvla resources."
+                                       :termsOfService "https://docs.nuvla.io/legal/legal/"
+                                       :contact        {:name  "API Support",
+                                                        :email "apiteam@nuvla.io"}
+                                       :license        {:name "Apache 2.0",
+                                                        :url  "https://www.apache.org/licenses/LICENSE-2.0.html"}
+                                       ;; TODO: add the api version
+                                       :version        "<version>"}
+                        :externalDocs {:description "Find out more about Nuvla"
+                                       :url         "https://docs.nuvla.io/nuvla/user-guide"}
+                        :servers      [{:url         "https://nuvla.io/api"
+                                        :description "Main (production) server"}]
+                        :tags         (tags resources-metadata)
+                        :paths        (paths resources-metadata)
+                        :components   {:schemas (schemas resources-metadata)}}]
+    #_(with-redefs [spec-tools.impl/extract-form
+                    (fn [spec]
+                      (if (seq? spec) spec (openapi-schema/->openapi-schema spec)))]
```

Comment:
I tried to use the openapi facilities offered by `spec-tools`, but I stopped using them after all: the added value seems to be not too much, some sections can be expressed more concisely but we lose some control. For example it made it more difficult to declare a custom parameter name for `resource-id`.
It seems easy enough to generate the spec on our own, and plug into `spec-tools` for schemas generation, so thats the approach taken here.